### ISSUE TITLE
Constructor changes for PHP7+ compatibility

### ISSUE THREAD

**Issue by APD183** [yodarunamok/fxphp#36]
Dipping my toes in PHP 7, it looks like FX.php needs some changes to its function constructors to operate error-free and continue to work for future versions. I can submit some changes later, but just wanted to point it out in case someone else has the time to submit changes.

> Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; FX has a deprecated constructor in C:\inetpub\www_test\html\includes\FX\FX.php on line 55

**Comment by yodarunamok** [comment]
Yep, one of the points of that development branch is to get FX.php all clean and modernized.  Things have been kinda crazy for me for a while, but I also wouldn't mind if someone(s) had time to help on this.  Documentation is also getting a really nice update.

**Comment by APD183** [comment]
What's the lowest version of PHP you're targeting for backward compatibility?

**Comment by yodarunamok** [comment]
There really needs to be another branch for this, as I'm still hoping to update things for people with older installs.  That said, PHP 5.3 would be nice, though I could be persuaded to use an even more recent version if there were a compelling reason.

**Comment by LRCUSUHS** [comment]
I just tried to upgrade to PHP 7.1 and I saw the same issues with the OO class constructors.  Is there any update on when a compatible FX will be available?  (By the way, thank you for keeping this code for everyone).  If it helps, our institution is willing to test the code.

**Comment by yodarunamok** [comment]
Okay, just created branches called `legacy` and `legacy-develop` from `master` which will be used for future fixes (as needed) for the old release. I also created a release based on the current state of `master` for easy download (version 6.10). From this point on, I'll be doing active development on the master branch based on compatibility with PHP7.

**Comment by rjakes** [comment]
Hi There!

I apologize if I'm missing something obvious, but, is there a version of FX that works with PHP 7+? I see that there was an intent to do so, but, it's not clear from the ReadMe.md, nor the branches, that FX is PHP 7+ ready yet.

Thanks!
